app/views/locations/_form.html.erb

```diff
@@ -0,0 +1,92 @@
+<%= form_with(model: location) do |form| %>
+  <% if @location.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(@location.errors.count, "error") %> prohibited this location from being saved:</h2>
+
+      <ul>
+        <% @location.errors.full_messages.each do |message| %>
+          <li><%= message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field">
+    <%= form.label :name %>
+    <%= form.text_field :name %>
+  </div>
+
+  <div class="field">
+    <%= form.label :details %>
+    <%= form.text_field :details %>
+  </div>
+
+  <div class="field">
+    <%= form.label :location_photo %>
+    <%= form.file_field :location_photo %>
+  </div>
+
+  <div class="field">
+    <%= form.label :location_latitude %>
+    <%= form.number_field :location_latitude, step: :any %>
+  </div>
+
+  <div class="field">
+    <%= form.label :location_longitude %>
+    <%= form.number_field :location_longitude, step: :any %>
+  </div>
+
+  <div class="field mb-3 opening-times-container">
+    <%= form.label :openingtimes, class: 'form-label' %>
+    <%= form.fields_for :openingtimes do |o| %>
+      <div class="field" id="opening-time-field-<%= o.object.id %>">
+        <%= o.select 'day',
+                     options_for_select({ 'Monday' => 0,
+                                          'Tuesday' => 1,
+                                          'Wednesday' => 2,
+                                          'Thursday' => 3,
+                                          'Friday' => 4,
+                                          'Saturday' => 5,
+                                          'Sunday' => 6
+                                        }, o.object.day),

```

Comment:
Worth a method in OpeningTimes? Weird that the other way around is a method but this not

Reply:
Good idea. Will do this.